pkg/core/blockchain.go

```diff
@@ -1388,13 +1390,14 @@ func (bc *Blockchain) processTokenTransfer(cache *dao.Simple, transCache map[uti
 		nep17xfer = &nep11xfer.NEP17Transfer
 	}
 	if !from.Equals(util.Uint160{}) {
-		_ = nep17xfer.Amount.Neg(amount) // We already have the Int.
-		if appendTokenTransfer(cache, transCache, from, transfer, id, b.Index, b.Timestamp, isNEP11) != nil {
+		_ = nep17xfer.Amount.Neg(&nep17xfer.Amount)
+		err := appendTokenTransfer(cache, transCache, from, transfer, id, b.Index, b.Timestamp, isNEP11)
+		_ = nep17xfer.Amount.Neg(&nep17xfer.Amount)
```

Comment:
Why should we do it one more time?

Reply:
Because we need the original value down below in the next `if`.